experiment/dispatcher.py

```diff
@@ -75,22 +67,44 @@ class Experiment:
     def __init__(self, experiment_config_filepath: str):
         self.config = yaml_utils.read(experiment_config_filepath)
 
-        benchmarks = self.config['benchmarks'].split(',')
-        self.benchmarks = builder.build_all_measurers(benchmarks)
+        self.benchmarks = self.config['benchmarks'].split(',')
 
         self.fuzzers = [
             fuzzer_config_utils.get_fuzzer_name(filename) for filename in
             os.listdir(fuzzer_config_utils.get_fuzzer_configs_dir())
         ]
-
-        _initialize_experiment_in_db(self.config['experiment'],
-                                     self.config['git_hash'], self.benchmarks,
-                                     self.fuzzers, self.config['trials'])
+        self.num_trials = self.config['trials']
+        self.experiment_name = self.config['experiment']
+        self.git_hash = self.config['git_hash']
 
         self.web_bucket = posixpath.join(self.config['cloud_web_bucket'],
                                          experiment_utils.get_experiment_name())
 
 
+def build_for_trials(fuzzers: List[str], benchmarks: List[str],
+                     num_trials: int) -> List[models.Trial]:
+    """Builds the images needed to run |experiment| and returns a list of trials
+    that can be run for experiment. This is the number of trials specified in
+    experiment times each pair of fuzzer+benchmark that builds successfully."""
+    # This call will raise an exception if the images can't be built which will
+    # halt the experiment.
+    builder.build_base_images()
+
+    # Only build fuzzers for benchmarks whose measurers built successfully.
+    benchmarks = builder.build_all_measurers(benchmarks)
+    build_successes = builder.build_all_fuzzer_benchmarks(fuzzers, benchmarks)
+    experiment_name = experiment_utils.get_experiment_name()
+    trials = []
+    for fuzzer, benchmark in build_successes:
```

Comment:
Do we log_error when things fail ? i think we should remember to check these after an experimetn run.

Reply:
We do log these.